crates/starknet-devnet/tests/test_blocks_on_demand.rs

```diff
@@ -0,0 +1,85 @@
+pub mod common;
+
+mod blocks_on_demand_tests {
+
+    use starknet_rs_core::types::{BlockStatus, FieldElement};
+    use starknet_types::rpc::transaction_receipt::FeeUnit;
+
+    use crate::common::background_devnet::BackgroundDevnet;
+    use crate::common::utils::assert_tx_successful;
+
+    static DUMMY_ADDRESS: u128 = 1;
+    static DUMMY_AMOUNT: u128 = 1;
+    const TX_COUNT: u128 = 5;
+
+    async fn assert_latest_block_with_transactions(
+        devnet: &BackgroundDevnet,
+        block_number: u64,
+        transactions_count: u128,
+    ) {
+        let latest_block = devnet.get_latest_block_with_tx_hashes().await.unwrap();
+        assert_eq!(latest_block.block_number, block_number);
+        assert_eq!(latest_block.transactions.len() as u128, transactions_count);
+        assert_eq!(latest_block.status, BlockStatus::AcceptedOnL2);
+
+        for tx_hash in latest_block.transactions {
+            assert_tx_successful(&tx_hash, &devnet.json_rpc_client).await;
+        }
+    }
+
+    async fn assert_pending_block_with_transactions(
+        devnet: &BackgroundDevnet,
+        block_number: u64,
+        transactions_count: u128,
+    ) {
+        let pending_block = devnet.get_pending_block_with_tx_hashes().await.unwrap();
+        assert_eq!(pending_block.block_number, block_number);
+        assert_eq!(pending_block.transactions.len() as u128, transactions_count);
+        assert_eq!(pending_block.status, BlockStatus::Pending);
+    }
+
+    async fn assert_balance(
+        devnet: &BackgroundDevnet,
+        expected: FieldElement,
+        pending_state: bool,
+    ) {
+        let balance = devnet
+            .get_balance_pending_state(
+                &FieldElement::from_hex_be(DUMMY_ADDRESS.to_string().as_str()).unwrap(),
+                FeeUnit::WEI,
+                pending_state,
+            )
+            .await
+            .unwrap();
+        assert_eq!(balance, expected);
+    }
+
+    #[tokio::test]
+    async fn blocks_on_demand_states_and_blocks() {
+        let devnet: BackgroundDevnet =
+            BackgroundDevnet::spawn_with_additional_args(&["--blocks-on-demand"])
+                .await
+                .expect("Could not start Devnet");
+
+        for _ in 0..TX_COUNT {
```

Comment:
I think this tx_count should be a local variable defined inside the test, not that I have to scroll to the top of the file to see it.

Reply:
43cfd9b0e2946fc54f7d4c073bc26c201087fb0f